fem/libceed/mf_integrator.hpp

```diff
@@ -0,0 +1,348 @@
+         case CeedCoeff::VecQuad:
+            qf = qf_file + op.vec_quad_func;
+            CeedQFunctionCreateInterior(ceed, 1, op.vec_quad_qf, qf.c_str(),
+                                        &apply_qfunc);
+            CeedQFunctionAddInput(apply_qfunc, "coeff", dim, CEED_EVAL_NONE);
+            break;
+      }
+      // input
+      switch (op.trial_op)
+      {
+         case EvalMode::None:
+            CeedQFunctionAddInput(apply_qfunc, "u", vdim, CEED_EVAL_NONE);
+            break;
+         case EvalMode::Interp:
+            CeedQFunctionAddInput(apply_qfunc, "u", vdim, CEED_EVAL_INTERP);
+            break;
+         case EvalMode::Grad:
+            CeedQFunctionAddInput(apply_qfunc, "gu", vdim*dim, CEED_EVAL_GRAD);
+            break;
+         case EvalMode::InterpAndGrad:
+            CeedQFunctionAddInput(apply_qfunc, "u", vdim, CEED_EVAL_INTERP);
+            CeedQFunctionAddInput(apply_qfunc, "gu", vdim*dim, CEED_EVAL_GRAD);
+            break;
+      }
+      CeedQFunctionAddInput(apply_qfunc, "dx", dim * dim, CEED_EVAL_GRAD);
+      CeedQFunctionAddInput(apply_qfunc, "weights", 1, CEED_EVAL_WEIGHT);
+      // output
+      switch (op.test_op)
+      {
+         case EvalMode::None:
+            CeedQFunctionAddOutput(apply_qfunc, "v", vdim, CEED_EVAL_NONE);
+            break;
+         case EvalMode::Interp:
+            CeedQFunctionAddOutput(apply_qfunc, "v", vdim, CEED_EVAL_INTERP);
+            break;
+         case EvalMode::Grad:
+            CeedQFunctionAddOutput(apply_qfunc, "gv", vdim*dim, CEED_EVAL_GRAD);
+            break;
+         case EvalMode::InterpAndGrad:
+            CeedQFunctionAddOutput(apply_qfunc, "v", vdim, CEED_EVAL_INTERP);
+            CeedQFunctionAddOutput(apply_qfunc, "gv", vdim*dim, CEED_EVAL_GRAD);
+            break;
+      }
+
+      CeedQFunctionContextCreate(ceed, &build_ctx);
+      CeedQFunctionContextSetData(build_ctx, CEED_MEM_HOST,
+                                  CEED_COPY_VALUES,
+                                  sizeof(ctx),
+                                  &ctx);
+      CeedQFunctionSetContext(apply_qfunc, build_ctx);
+
+      // Create the operator.
+      CeedOperatorCreate(ceed, apply_qfunc, NULL, NULL, &oper);
+      // coefficient
+      switch (coeff_type)
+      {
+         case CeedCoeff::Const:
+            break;
+         case CeedCoeff::Grid:
+         {
+            CeedGridCoeff* gridCoeff = (CeedGridCoeff*)coeff;
+            InitCeedBasisAndRestriction(*gridCoeff->coeff->FESpace(), irm, ceed,
+                                        &gridCoeff->basis, &gridCoeff->restr);
+            CeedOperatorSetField(oper, "coeff", gridCoeff->restr,
+                                 gridCoeff->basis, gridCoeff->coeffVector);
+         }
+         break;
+         case CeedCoeff::Quad:
+         {
+            CeedQuadCoeff* quadCoeff = (CeedQuadCoeff*)coeff;
+            const int ncomp = 1;
+            CeedInt strides[3] = {1, nqpts, ncomp*nqpts};
```

Comment:
Should it be `{1, 1, ncomp*nqpts};`?

Reply:
This is the same, there is only one component, the component stride is not used. But if the layout is `[comp, quad, elem]`, then  what you wrote is a bit more correct, or `{ncomp, 1, ncomp*nqpts}`, or `{1, 0, nqpts}`, or `{1,1,nqpts}`.